### PR TITLE
Broadcast case id on case create/update events

### PR DESCRIPTION
**Pull request by phillipm**
This is needed so that Simprints knows

- when they should mark fingerprint registration as committed.
- when a case is selected, allowing them to refine their matching algorithm

cross-request: https://github.com/dimagi/commcare-core/pull/329

### CODE REVIEW

File: app/src/org/commcare/activities/EntityDetailActivity.java [hunk 1]
```diff
@@ -224,6 +224,11 @@ protected boolean onBackwardSwipe() {
      * Move along to form entry.
      */
     private void select() {
+        // announce that case was selected
+        Intent selectIntentBroadcast = new Intent("org.commcare.dalvik.api.action.case.selected");
```
Comment (amstone326) [inline app/src/org/commcare/activities/EntityDetailActivity.java]: could move this into a well-named helper and remove the need for an explanatory comment

File: app/src/org/commcare/models/FormRecordProcessor.java [hunk 2]
```diff
@@ -85,7 +85,8 @@ public TransactionParser getParser(KXmlParser parser) {
 
         //Let anyone who is listening know!
         Intent i = new Intent("org.commcare.dalvik.api.action.data.update");
-        this.c.sendBroadcast(i);
+        i.putStringArrayListExtra("cases", factory.getCreatedAndUpdatedCases());
```
Comment (amstone326) [inline app/src/org/commcare/models/FormRecordProcessor.java]: is this definitely something that's ok to be broadcasting?

Comment (phillipm) [inline app/src/org/commcare/models/FormRecordProcessor.java]: As far as I know this seems fine. @ctsims any opinions?

**Comment by amstone326** [comment]
@phillipm this is still failing after triggering a rebuild on jenkins, would suggest maybe making a noop change and pushing to re-trigger from here?

File: app/src/org/commcare/models/FormRecordProcessor.java (hunk 2 above)
Comment (ctsims) [inline app/src/org/commcare/models/FormRecordProcessor.java]: Interesting... We do treat case Id's as private data generally. IS it clear whether this broadcast is permissioned? There's an Android "read CommCare data" permission that we require for the content provider. If it's set for the broadcasts as well this seems good to go.

Comment (phillipm) [inline app/src/org/commcare/models/FormRecordProcessor.java]: Ah, good call. I've added the case read permission and an example of it being read in https://github.com/dimagi/commcare-tester-app/pull/4

**Pull request merged**